src/dynspec/doc/ICD6-ICD6-Rebin/man/man3/Stock_Write_Dyn2Dyn_Metadata_Part.3
.TH "Stock_Write_Dyn2Dyn_Metadata_Part" 3 "Thu Jan 10 2013" "LOFAR-ICD6 Convertor with selection and/or rebinning" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Stock_Write_Dyn2Dyn_Metadata_Part \- 
.PP
Class object for stocking and writing metadatas for the ICD6's Dynspec Groups (loop on Beams \&.\&.\&.)  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Stock_Write_Dyn2Dyn_Metadata_Part\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "void \fBstockDynspecMetadata\fP (std::string attr_69, double attr_70, double attr_71, std::string attr_72, std::string attr_73, double attr_76, double attr_77, std::string attr_78, std::string attr_79, std::string attr_80, double attr_81, double attr_82, double attr_83, double attr_84, double attr_85, double attr_86, double attr_87, double attr_88, double attr_89, std::string attr_90, double attr_91, std::vector< std::string > attr_92, std::string attr_93, double attr_94, std::string attr_95, bool attr_96, std::vector< std::string > attr_97, bool attr_98, std::string attr_99, std::string attr_100, std::vector< double > attr_101, std::vector< std::string > attr_102, std::string attr_103, double attr_104, std::string attr_105, std::string attr_106, double attr_107, double attr_108, std::vector< std::string > attr_109, std::string attr_110, std::string attr_111, std::vector< std::string > attr_112, int attr_113, std::vector< std::string > attr_114, std::vector< std::string > attr_115, std::vector< double > attr_116, std::vector< double > attr_117, std::vector< double > attr_118, std::vector< double > attr_119, std::string attr_123, std::string attr_124, std::vector< std::string > attr_125, int attr_126, std::vector< std::string > attr_127, std::vector< std::string > attr_128, std::vector< double > attr_129, std::vector< double > attr_130, std::vector< double > attr_131, std::vector< double > attr_132, std::vector< double > attr_133, std::vector< double > attr_135, std::string attr_137, std::string attr_138, std::vector< std::string > attr_139, int attr_140, std::vector< std::string > attr_141, std::vector< std::string > attr_142, int Ntime, int Nspectral, std::string attr_147, std::string attr_148, int attr_149, std::string attr_150, std::string attr_151, int attr_152, std::vector< std::string > attr_153, std::vector< std::string > attr_154, std::vector< std::string > attr_155, std::vector< std::string > attr_156, std::vector< std::string > attr_157, std::vector< std::string > attr_158, std::string attr_159, bool attr_160, bool attr_161, bool attr_162, bool attr_163, std::string attr_164, std::string attr_165, std::string attr_166, std::string attr_167, std::string attr_168)"
.br
.ti -1c
.RI "void \fBwriteDynspecMetadata\fP (Group &dynspec_grp, std::string obsName, std::string pathFile, std::string outputFile, File &root_grp, int j, int k, int obsNofStockes, std::vector< std::string > stokesComponent, float memoryRAM)"
.br
.in -1c
.SH "Detailed Description"
.PP 
Class object for stocking and writing metadatas for the ICD6's Dynspec Groups (loop on Beams \&.\&.\&.) 


.br
 Usage: 
.br
 This class need as parameter all metadata for the ICD6 (or dynamic spectrum)'s Dynspec Groups\&. The function stockDynspecMetadata stocks dynspec metadata in private attributes, and the function writeDynspecMetadata write them in the corresponding dynamic spectrum Group of the hdf5 file in output 
.SH "Member Function Documentation"
.PP 
.SS "void \fBStock_Write_Dyn2Dyn_Metadata_Part::stockDynspecMetadata\fP (std::stringattr_69, doubleattr_70, doubleattr_71, std::stringattr_72, std::stringattr_73, doubleattr_76, doubleattr_77, std::stringattr_78, std::stringattr_79, std::stringattr_80, doubleattr_81, doubleattr_82, doubleattr_83, doubleattr_84, doubleattr_85, doubleattr_86, doubleattr_87, doubleattr_88, doubleattr_89, std::stringattr_90, doubleattr_91, std::vector< std::string >attr_92, std::stringattr_93, doubleattr_94, std::stringattr_95, boolattr_96, std::vector< std::string >attr_97, boolattr_98, std::stringattr_99, std::stringattr_100, std::vector< double >attr_101, std::vector< std::string >attr_102, std::stringattr_103, doubleattr_104, std::stringattr_105, std::stringattr_106, doubleattr_107, doubleattr_108, std::vector< std::string >attr_109, std::stringattr_110, std::stringattr_111, std::vector< std::string >attr_112, intattr_113, std::vector< std::string >attr_114, std::vector< std::string >attr_115, std::vector< double >attr_116, std::vector< double >attr_117, std::vector< double >attr_118, std::vector< double >attr_119, std::stringattr_123, std::stringattr_124, std::vector< std::string >attr_125, intattr_126, std::vector< std::string >attr_127, std::vector< std::string >attr_128, std::vector< double >attr_129, std::vector< double >attr_130, std::vector< double >attr_131, std::vector< double >attr_132, std::vector< double >attr_133, std::vector< double >attr_135, std::stringattr_137, std::stringattr_138, std::vector< std::string >attr_139, intattr_140, std::vector< std::string >attr_141, std::vector< std::string >attr_142, intNtime, intNspectral, std::stringattr_147, std::stringattr_148, intattr_149, std::stringattr_150, std::stringattr_151, intattr_152, std::vector< std::string >attr_153, std::vector< std::string >attr_154, std::vector< std::string >attr_155, std::vector< std::string >attr_156, std::vector< std::string >attr_157, std::vector< std::string >attr_158, std::stringattr_159, boolattr_160, boolattr_161, boolattr_162, boolattr_163, std::stringattr_164, std::stringattr_165, std::stringattr_166, std::stringattr_167, std::stringattr_168)"
.br
 Usage: 
.br
 void \fBStock_Write_Dyn2Dyn_Metadata_Part::stockDynspecMetadata\fP(string attr_69,string attr_70,string attr_71,string attr_72,string attr_73,double attr_76,double attr_77,string attr_78,string attr_79, string attr_80,double attr_81,double attr_82,double attr_83,double attr_84,double attr_85,double attr_86,double attr_87,double attr_88,double attr_89, string attr_90,double attr_91,vector<string> attr_92,string attr_93,double attr_94,string attr_95,bool attr_96,vector<string> attr_97,bool attr_98,string attr_99, string attr_100,vector<double> attr_101,vector<string> attr_102,string attr_103,double attr_104,string attr_105,string attr_106,double attr_107,double attr_108,vector<string> attr_109, string attr_110,string attr_111,vector<string> attr_112,int attr_113,vector<string> attr_114,vector<string> attr_115,vector<double> attr_116,vector<double> attr_117,vector<double> attr_118, vector<double> attr_119,string attr_123,string attr_124,vector<string> attr_125,int attr_126,vector<string> attr_127, vector<string> attr_128,vector<double> attr_129,vector<double> attr_130,vector<double> attr_131,vector<double> attr_132,vector<double> attr_133,vector<double> attr_135, string attr_137,string attr_138,vector<string> attr_139,int attr_140,vector<string> attr_141,vector<string> attr_142,int Ntime,int Nspectral,string attr_147,string attr_148,int attr_149,string attr_150, string attr_151,int attr_152,vector<string> attr_153,vector<string> attr_154,vector<string> attr_155,vector<string> attr_156,vector<string> attr_157,vector<string> attr_158,string attr_159,bool attr_160,bool attr_161,bool attr_162,bool attr_163, string attr_164,string attr_165,string attr_166,string attr_167,string attr_168)
.PP
\fBParameters:\fP
.RS 4
\fIattr_69\fP GROUPE_TYPE_DYN 
.br
\fIattr_70\fP DYNSPEC_START_MJD 
.br
\fIattr_71\fP DYNSPEC_STOP_MJD 
.br
\fIattr_72\fP DYNSPEC_START_UTC 
.br
\fIattr_73\fP DYNSPEC_STOP_UTC 
.br
\fIattr_76\fP DYNSPEC_BANDWIDTH 
.br
\fIattr_77\fP BEAM_DIAMETER 
.br
\fIattr_78\fP TRACKING 
.br
\fIattr_79\fP TARGET_DYN 
.br
\fIattr_80\fP ONOFF 
.br
\fIattr_81\fP POINT_RA 
.br
\fIattr_82\fP POINT_DEC 
.br
\fIattr_83\fP POSITION_OFFSET_RA 
.br
\fIattr_84\fP POSITION_OFFSET_DEC 
.br
\fIattr_85\fP BEAM_DIAMETER_RA 
.br
\fIattr_86\fP BEAM_DIAMETER_DEC 
.br
\fIattr_87\fP BEAM_FREQUENCY_MAX 
.br
\fIattr_88\fP BEAM_FREQUENCY_MIN 
.br
\fIattr_89\fP BEAM_FREQUENCY_CENTER 
.br
\fIattr_90\fP BEAM_FREQUENCY_UNIT 
.br
\fIattr_91\fP BEAM_NOF_STATIONS 
.br
\fIattr_92\fP BEAM_STATIONS_LIST 
.br
\fIattr_93\fP DEDISPERSION 
.br
\fIattr_94\fP DISPERSION_MEASURE 
.br
\fIattr_95\fP DISPERSION_MEASURE_UNIT 
.br
\fIattr_96\fP BARYCENTER 
.br
\fIattr_97\fP STOCKES_COMPONENT 
.br
\fIattr_98\fP COMPLEX_VOLTAGE 
.br
\fIattr_99\fP SIGNAL_SUM 
.br
\fIattr_100\fP GROUPE_TYPE_COORD 
.br
\fIattr_101\fP REF_LOCATION_VALUE 
.br
\fIattr_102\fP REF_LOCATION_UNIT 
.br
\fIattr_103\fP REF_LOCATION_FRAME 
.br
\fIattr_104\fP REF_TIME_VALUE 
.br
\fIattr_105\fP REF_TIME_UNIT 
.br
\fIattr_106\fP REF_TIME_FRAME 
.br
\fIattr_107\fP NOF_COORDINATES 
.br
\fIattr_108\fP NOF_AXIS 
.br

src/dynspec/src/ICD6-ICD6-Rebin/Stock_Write_Dyn2Dyn_Metadata_Part.cpp
#include <iostream>
#include <string>

#include "Stock_Write_Dyn2Dyn_Metadata_Part.h"

#include <dal/lofar/BF_File.h>

using namespace dal;

using namespace std;


  Stock_Write_Dyn2Dyn_Metadata_Part::Stock_Write_Dyn2Dyn_Metadata_Part(){}
  Stock_Write_Dyn2Dyn_Metadata_Part::~Stock_Write_Dyn2Dyn_Metadata_Part(){}


  void Stock_Write_Dyn2Dyn_Metadata_Part::stockDynspecMetadata(string attr_69,double attr_70,double attr_71,string attr_72,string attr_73,double attr_76,double attr_77,string attr_78,string attr_79,
			      string attr_80,double attr_81,double attr_82,double attr_83,double attr_84,double attr_85,double attr_86,double attr_87,double attr_88,double attr_89,
			      string attr_90,double attr_91,vector<string> attr_92,string attr_93,double attr_94,string attr_95,bool attr_96,vector<string> attr_97,bool attr_98,string attr_99,
			      string attr_100,vector<double> attr_101,vector<string> attr_102,string attr_103,double attr_104,string attr_105,string attr_106,double attr_107,double attr_108,vector<string> attr_109,
			      string attr_110,string attr_111,vector<string> attr_112,int attr_113,vector<string> attr_114,vector<string> attr_115,vector<double> attr_116,vector<double> attr_117,vector<double> attr_118,
			      vector<double> attr_119,string attr_123,string attr_124,vector<string> attr_125,int attr_126,vector<string> attr_127,
			      vector<string> attr_128,vector<double> attr_129,vector<double> attr_130,vector<double> attr_131,vector<double> attr_132,vector<double> attr_133,vector<double> attr_135,
			      string attr_137,string attr_138,vector<string> attr_139,int attr_140,vector<string> attr_141,vector<string> attr_142,int Ntime,int Nspectral,string attr_147,string attr_148,int attr_149,string attr_150,
			      string attr_151,int attr_152,vector<string> attr_153,vector<string> attr_154,vector<string> attr_155,vector<string> attr_156,vector<string> attr_157,vector<string> attr_158,string attr_159,bool attr_160,bool attr_161,bool attr_162,bool attr_163,
			      string attr_164,string attr_165,string attr_166,string attr_167,string attr_168)
    
 {
   // Stock DYNSPEC metadata in an object
   
    // DYNSPEC group meta-data
    
    m_attr_69	= attr_69;
    m_attr_70	= attr_70;
    m_attr_71	= attr_71;
    m_attr_72	= attr_72;
    m_attr_73	= attr_73;
    m_attr_76	= attr_76;
    m_attr_77	= attr_77;
    m_attr_78	= attr_78;
    m_attr_79 	= attr_79;    
    m_attr_80	= attr_80;
    m_attr_81	= attr_81;
    m_attr_82	= attr_82;
    m_attr_83	= attr_83;
    m_attr_84	= attr_84;
    m_attr_85	= attr_85;
    m_attr_86	= attr_86;
    m_attr_87	= attr_87;
    m_attr_88	= attr_88;
    m_attr_89	= attr_89;
    m_attr_90	= attr_90;
    m_attr_91	= attr_91;
 
    m_size_attr_92= attr_92.size();
    m_attr_92.resize(m_size_attr_92);
    for (size_t i=0;i< attr_92.size();i++)
    {m_attr_92[i] = attr_92[i];}
 
    m_attr_93	= attr_93;
    m_attr_94	= attr_94;
    m_attr_95	= attr_95;
    m_attr_95	= attr_96;
    
    m_size_attr_97= attr_97.size();
    m_attr_97.resize(m_size_attr_97);
    for (size_t i=0;i< attr_97.size();i++)
    {m_attr_97[i] = attr_97[i];}
 
    m_attr_98	= attr_98;
    m_attr_99	= attr_99; 
    m_attr_100	= attr_100;
    
    m_size_attr_101= attr_101.size();
    m_attr_101.resize(m_size_attr_101);
    for (size_t i=0;i< attr_101.size();i++)
    {m_attr_101[i] = attr_101[i];}
    
    m_size_attr_102= attr_102.size();
    m_attr_102.resize(m_size_attr_102);
    for (size_t i=0;i< attr_102.size();i++)
    {m_attr_102[i] = attr_102[i];}

    m_attr_103	= attr_103;
    m_attr_104	= attr_104;
    m_attr_105	= attr_105;
    m_attr_106	= attr_106;
    m_attr_107	= attr_107;
    m_attr_108	= attr_108;
      
    m_size_attr_109= attr_109.size();
    m_attr_109.resize(m_size_attr_109);
    for (size_t i=0;i< attr_109.size();i++)
    {m_attr_109[i] = attr_109[i];}
    
    
    m_attr_110	= attr_110;
    m_attr_111	= attr_111;
    
    m_size_attr_112= attr_112.size();
    m_attr_112.resize(m_size_attr_112);
    for (size_t i=0;i< attr_112.size();i++)
    {m_attr_112[i] = attr_112[i];}
    
    m_attr_113	= attr_113;
    
    m_size_attr_114= attr_114.size();
    m_attr_114.resize(m_size_attr_114);
    for (size_t i=0;i< attr_114.size();i++)
    {m_attr_114[i] = attr_114[i];}
    
    m_size_attr_115= attr_115.size();
    m_attr_115.resize(m_size_attr_115);
    for (size_t i=0;i< attr_115.size();i++)
    {m_attr_115[i] = attr_115[i];}
    
    m_size_attr_116= attr_116.size();
    m_attr_116.resize(m_size_attr_116);
    for (size_t i=0;i< attr_116.size();i++)
    {m_attr_116[i] = attr_116[i];}
    
    m_size_attr_117= attr_117.size();
    m_attr_117.resize(m_size_attr_117);
    for (size_t i=0;i< attr_117.size();i++)
    {m_attr_117[i] = attr_117[i];}
    
    m_size_attr_118= attr_118.size();
    m_attr_118.resize(m_size_attr_118);
    for (size_t i=0;i< attr_118.size();i++)
    {m_attr_118[i] = attr_118[i];}
    
    m_size_attr_119= attr_119.size();
    m_attr_119.resize(m_size_attr_119);
    for (size_t i=0;i< attr_119.size();i++)
    {m_attr_119[i] = attr_119[i];}
			        

    m_attr_123	= attr_123;
    m_attr_124	= attr_124;
    
    m_size_attr_125= attr_125.size();
    m_attr_125.resize(m_size_attr_125);
    for (size_t i=0;i< attr_125.size();i++)
    {m_attr_125[i] = attr_125[i];}
    
    m_attr_126	= attr_126;
    
    m_size_attr_127= attr_127.size();
    m_attr_127.resize(m_size_attr_127);
    for (size_t i=0;i< attr_127.size();i++)
    {m_attr_127[i] = attr_127[i];}
    
    m_size_attr_128= attr_128.size();
    m_attr_128.resize(m_size_attr_128);
    for (size_t i=0;i< attr_128.size();i++)
    {m_attr_128[i] = attr_128[i];}
    
    m_size_attr_129= attr_129.size();
    m_attr_129.resize(m_size_attr_129);
    for (size_t i=0;i< attr_129.size();i++)
    {m_attr_129[i] = attr_129[i];}
    
    m_size_attr_130= attr_130.size();
    m_attr_130.resize(m_size_attr_130);
    for (size_t i=0;i< attr_130.size();i++)
    {m_attr_130[i] = attr_130[i];}
    
    m_size_attr_131= attr_131.size();
    m_attr_131.resize(m_size_attr_131);
    for (size_t i=0;i< attr_131.size();i++)
    {m_attr_131[i] = attr_131[i];}
    
    m_size_attr_132= attr_132.size();
    m_attr_132.resize(m_size_attr_132);
    for (size_t i=0;i< attr_132.size();i++)
    {m_attr_132[i] = attr_132[i];}
    
    m_size_attr_133= attr_133.size();
    m_attr_133.resize(m_size_attr_133);
    for (size_t i=0;i< attr_133.size();i++)
    {m_attr_133[i] = attr_133[i];}
    
    m_size_attr_135= attr_135.size();
    m_attr_135.resize(m_size_attr_135);
    for (size_t i=0;i< attr_135.size();i++)
    {m_attr_135[i] = attr_135[i];}

    m_attr_137	= attr_137;
    m_attr_138	= attr_138;
    
    m_size_attr_139= attr_139.size();
    m_attr_139.resize(m_size_attr_139);
    for (size_t i=0;i< attr_139.size();i++)
    {m_attr_139[i] = attr_139[i];}
    
    m_attr_140	= attr_140;
    
    m_size_attr_141= attr_141.size();
    m_attr_141.resize(m_size_attr_141);
    for (size_t i=0;i< attr_141.size();i++)
    {m_attr_141[i] = attr_141[i];}
    
    m_size_attr_142= attr_142.size();
    m_attr_142.resize(m_size_attr_142);
    for (size_t i=0;i< attr_142.size();i++)
    {m_attr_142[i] = attr_142[i];}
    
    m_Ntime	= Ntime;
    m_Nspectral	= Nspectral;
    m_attr_147	= attr_147;
    m_attr_148	= attr_148;
    m_attr_149	= attr_149;
    m_attr_150	= attr_150;
    m_attr_151	= attr_151;
    m_attr_152	= attr_152;
    
    m_size_attr_153= attr_153.size();
    m_attr_153.resize(m_size_attr_153);
    for (size_t i=0;i< attr_153.size();i++)
    {m_attr_153[i] = attr_153[i];}   
        
    m_size_attr_154= attr_154.size();
    m_attr_154.resize(m_size_attr_154);
    for (size_t i=0;i< attr_154.size();i++)
    {m_attr_154[i] = attr_154[i];}       
    
    m_size_attr_155= attr_155.size();
    m_attr_155.resize(m_size_attr_155);
    for (size_t i=0;i< attr_155.size();i++)
    {m_attr_155[i] = attr_155[i];}    
    
    m_size_attr_156= attr_156.size();
    m_attr_156.resize(m_size_attr_156);
    for (size_t i=0;i< attr_156.size();i++)
    {m_attr_156[i] = attr_156[i];}    
    
    m_size_attr_157= attr_157.size();
    m_attr_157.resize(m_size_attr_157);
    for (size_t i=0;i< attr_157.size();i++)
    {m_attr_157[i] = attr_157[i];}    
    
    m_size_attr_158= attr_158.size();
    m_attr_158.resize(m_size_attr_158);
    for (size_t i=0;i< attr_158.size();i++)
    {m_attr_158[i] = attr_158[i];}    
    
    m_attr_159	= attr_159;
    m_attr_160	= attr_160;
    m_attr_161	= attr_161;
    m_attr_162	= attr_162;
    m_attr_163	= attr_163;
    
    m_attr_164	= attr_164;  
    m_attr_165	= attr_165;  
    m_attr_166	= attr_166;  
    m_attr_167	= attr_167;  
    m_attr_168	= attr_168;  
    
  }


////////////////////////////////////////////////////////////////////////////////////////////////////////


  void Stock_Write_Dyn2Dyn_Metadata_Part::writeDynspecMetadata(Group &dynspec_grp,string obsName,string pathFile,string outputFile,File &root_grp,int j,int k,int obsNofStockes,vector<string> stokesComponent,float memoryRAM)
  {
    
      // Write the DYNSPEC meta-data in the output file

      string index_j1;     
      std::ostringstream oss_j;oss_j << j;string index_j(oss_j.str()); 
      if (j<10){index_j1="_00"+index_j;}
      if (j>=10 && j<100){index_j1="_0"+index_j;}
      if (j>=100 && j<1000){index_j1="_"+index_j;}      

      
      /////////////////////////////////////////////////
     
      // SUB GROUPS
      Group coords_grp(dynspec_grp, "COORDINATES");
      coords_grp.create();
      //SUB GROUP EVENT
      Group event_grp(dynspec_grp, "EVENT");
      event_grp.create();

       //SUB GROUP EVENT      
      Group process_histo_grp(dynspec_grp, "PROCESS_HISTORY");           
      process_histo_grp.create();   

      //SUBSUB GROUP 
      Group time_grp(coords_grp, "TIME");
      time_grp.create();
      
      Group spectral_grp(coords_grp, "SPECTRAL");
      spectral_grp.create();       

      Group polarization_grp(coords_grp, "POLARIZATION");
      polarization_grp.create(); 
      
      
      // META DATA
      
      
      // DYNSPEC GROUP

      Attribute<string> attr_69(dynspec_grp, "GROUPETYPE");
      Attribute<string> attr_70(dynspec_grp, "DYNSPEC_START_MJD");
      Attribute<string> attr_71(dynspec_grp, "DYNSPEC_STOP_MJD");
      Attribute<string> attr_72(dynspec_grp, "DYNSPEC_START_UTC");
      Attribute<string> attr_73(dynspec_grp, "DYNSPEC_STOP_UTC");
      Attribute<double> attr_76(dynspec_grp, "DYNSPEC_BANDWIDTH");
      Attribute<double> attr_77(dynspec_grp, "BEAM_DIAMETER");     
      Attribute<string> attr_78(dynspec_grp, "TRACKING");
      Attribute<string> attr_79(dynspec_grp, "TARGET");
      Attribute<string> attr_80(dynspec_grp, "ONOFF");     
      Attribute<double> attr_81(dynspec_grp, "POINT_RA");
      Attribute<double> attr_82(dynspec_grp, "POINT_DEC");     
      Attribute<double> attr_83(dynspec_grp, "POSITION_OFFSET_RA");
      Attribute<double> attr_84(dynspec_grp, "POSITION_OFFSET_DEC");
      Attribute<double> attr_85(dynspec_grp, "BEAM_DIAMETER_RA");
      Attribute<double> attr_86(dynspec_grp, "BEAM_DIAMETER_DEC");     
      Attribute<double> attr_87(dynspec_grp, "BEAM_FREQUENCY_MAX");
      Attribute<double> attr_88(dynspec_grp, "BEAM_FREQUENCY_MIN");
      Attribute<double> attr_89(dynspec_grp, "BEAM_FREQUENCY_CENTER");
      Attribute<string> attr_90(dynspec_grp, "BEAM_FREQUENCY_UNIT");      
      Attribute<double> attr_91(dynspec_grp, "BEAM_NOF_STATIONS");
      Attribute< vector<string> > attr_92(dynspec_grp, "BEAM_STATIONS_LIST");     
      Attribute<string> attr_93(dynspec_grp, "DEDISPERSION");
      Attribute<double> attr_94(dynspec_grp, "DISPERSION_MEASURE");
      Attribute<string> attr_95(dynspec_grp, "DISPERSION_MEASURE_UNIT");
      Attribute<bool> attr_96(dynspec_grp, "BARYCENTER");
      Attribute< vector<string> > attr_97(dynspec_grp, "STOCKES_COMPONENT");
      Attribute<bool> attr_98(dynspec_grp, "COMPLEX_VOLTAGE");
      Attribute<string> attr_99(dynspec_grp, "SIGNAL_SUM");
      
      
      attr_69.value = m_attr_69;
      attr_70.value = m_attr_70;  
      attr_71.value = m_attr_71;
      attr_72.value = m_attr_72;
      attr_73.value = m_attr_73;
      attr_76.value = m_attr_76;
      attr_77.value = m_attr_77;
      attr_78.value = m_attr_78;
      attr_79.value = m_attr_79;   
      attr_80.value = m_attr_80;
      attr_81.value = m_attr_81;      
      attr_82.value = m_attr_82;
      attr_83.value = m_attr_83;
      attr_84.value = m_attr_84;
      attr_85.value = m_attr_85;
      attr_86.value = m_attr_86;
      attr_87.value = m_attr_87;
      attr_88.value = m_attr_88;
      attr_89.value = m_attr_89;      
      attr_90.value = m_attr_90;
      attr_91.value = m_attr_91;      
      attr_92.create().set(m_attr_92);    
      attr_93.value = m_attr_93;
      attr_94.value = m_attr_94;
      attr_95.value = m_attr_95;
      attr_96.value = m_attr_96;      
      attr_97.create().set(m_attr_97);     
      attr_98.value = m_attr_98;
      attr_99.value = m_attr_99;
            
      
      // SUB-GROUP COORDINATES
      
      Attribute<string> attr_100(coords_grp, "GROUPE_TYPE");
      Attribute< vector<double> > attr_101(coords_grp, "REF_LOCATION_VALUE");
      Attribute< vector<string> > attr_102(coords_grp, "REF_LOCATION_UNIT");
      Attribute<string> attr_103(coords_grp, "REF_LOCATION_FRAME");
      Attribute<double> attr_104(coords_grp, "REF_TIME_VALUE");
      Attribute<string> attr_105(coords_grp, "REF_TIME_UNIT");
      Attribute<string> attr_106(coords_grp, "REF_TIME_FRAME");
      Attribute<double> attr_107(coords_grp, "NOF_COORDINATES");
      Attribute<double> attr_108(coords_grp, "NOF_AXIS");
      Attribute< vector<string> > attr_109(coords_grp, "COORDINATE_TYPES");      
      
      attr_100.value = m_attr_100;      
      attr_101.create().set(m_attr_101);
      attr_102.create().set(m_attr_102);
      attr_103.value = m_attr_103;
      attr_104.value = m_attr_104;
      attr_105.value = m_attr_105;
      attr_106.value = m_attr_106;
      attr_107.value = m_attr_107;
      attr_108.value = m_attr_108;
      attr_109.create().set(m_attr_109);      
        
      
      //SUBGROUP EVENT

      Attribute<string> attr_147(event_grp,"GROUPE_TYPE");
      Attribute<string> attr_148(event_grp,"DATASET");
      Attribute<int> attr_149(event_grp,"N_AXIS");
      Attribute<string> attr_150(event_grp,"N_AXIS_1");
      Attribute<string> attr_151(event_grp,"N_AXIS_2");
      Attribute<int> attr_152(event_grp,"N_EVENT");
      Attribute< vector<string> >attr_153(event_grp,"FIELD_1");
      Attribute< vector<string> >attr_154(event_grp,"FIELD_2");
      Attribute< vector<string> >attr_155(event_grp,"FIELD_3");
      Attribute< vector<string> >attr_156(event_grp,"FIELD_4");
      Attribute< vector<string> >attr_157(event_grp,"FIELD_5");
      Attribute< vector<string> >attr_158(event_grp,"FIELD_6");
      
      attr_147.value	= m_attr_147;
      attr_148.value	= m_attr_148;
      attr_149.value	= m_attr_149;
      attr_150.value	= m_attr_150;
      attr_151.value	= m_attr_151;
      attr_152.value	= m_attr_152;
      attr_153.create().set(m_attr_153); 
      attr_154.create().set(m_attr_154);      
      attr_155.create().set(m_attr_155);   
      attr_156.create().set(m_attr_156);   
      attr_157.create().set(m_attr_157);   
      attr_158.create().set(m_attr_158);   

      
      // SUB GROUP PROCESS HISTO

      Attribute<string> attr_159(process_histo_grp,"GROUPE_TYPE");      
      Attribute<bool> attr_160(process_histo_grp,"OBSERVATION_PARSET");        
      Attribute<bool> attr_161(process_histo_grp,"OBSERVATION_LOG");
      Attribute<bool> attr_162(process_histo_grp,"DYNSPEC_PARSET");
      Attribute<bool> attr_163(process_histo_grp,"DYNSPEC_LOG");
      
      Attribute<string> attr_164(process_histo_grp,"EXTRACTION LOG1");
      Attribute<string> attr_165(process_histo_grp,"EXTRACTION LOG2");     
      Attribute<string> attr_166(process_histo_grp,"EXTRACTION LOG3");     
      Attribute<string> attr_167(process_histo_grp,"EXTRACTION LOG4");
      Attribute<string> attr_168(process_histo_grp,"EXTRACTION LOG5");
	     
      attr_159.value	= m_attr_159;
      attr_160.value	= m_attr_160;
      attr_161.value	= m_attr_161;
      attr_162.value	= m_attr_162;
      attr_163.value	= m_attr_163;
      
      attr_164.value	= m_attr_164;      
      attr_165.value	= m_attr_165;        
      attr_166.value	= m_attr_166;
      attr_167.value	= m_attr_167;
      attr_168.value	= m_attr_168;
      
      
      // SUB-SUB-SUBGROUPS
      
      //TIME

      Attribute<string> attr_110(time_grp, "GROUPE_TYPE");
      Attribute<string> attr_111(time_grp, "COORDINATE_TYPE");
      Attribute< vector<string> > attr_112(time_grp, "STORAGE_TYPE");
      Attribute<int> attr_113(time_grp, "NOF_AXES");
      Attribute< vector<string> > attr_114(time_grp, "AXIS_NAMES");
      Attribute< vector<string> > attr_115(time_grp, "AXIS_UNIT");      
      Attribute< vector<double> > attr_116(time_grp, "REFERENCE_VALUE");
      Attribute< vector<double> > attr_117(time_grp, "REFERENCE_PIXEL");
      Attribute< vector<double> > attr_118(time_grp, "INCREMENT");
      Attribute< vector<double> > attr_119(time_grp, "PC");          

      
      attr_110.value = m_attr_110;
      attr_111.value = m_attr_111;
      attr_112.create().set(m_attr_112);      
      attr_113.value = m_attr_113;
      attr_114.create().set(m_attr_114); 
      attr_115.create().set(m_attr_115); 
      attr_116.create().set(m_attr_116);
      attr_117.create().set(m_attr_117);
      attr_118.create().set(m_attr_118);
      attr_119.create().set(m_attr_119);      
     
      
      // SPECTRAL

      Attribute<string> attr_123(spectral_grp, "GROUPE_TYPE");
      Attribute<string> attr_124(spectral_grp, "COORDINATE_TYPE");
      Attribute< vector<string> > attr_125(spectral_grp, "STORAGE_TYPE");
      Attribute<int> attr_126(spectral_grp, "NOF_AXES");
      Attribute< vector<string> > attr_127(spectral_grp, "AXIS_NAMES");
      Attribute< vector<string> > attr_128(spectral_grp, "AXIS_UNIT");
      Attribute< vector<double> > attr_129(spectral_grp, "REFERENCE_VALUE");
      Attribute< vector<double> > attr_130(spectral_grp, "REFERENCE_PIXEL");
      Attribute< vector<double> > attr_131(spectral_grp, "INCREMENT");
      Attribute< vector<double> > attr_132(spectral_grp, "PC");
      Attribute< vector<double> > attr_133(spectral_grp, "AXIS_VALUE_WORLD");
      Attribute< vector<double> > attr_135(spectral_grp, "AXIS_VALUE_PIXEL");
      
      attr_123.value = m_attr_123;
      attr_124.value = m_attr_124;
      attr_125.create().set(m_attr_125);
      attr_126.value = m_attr_126;
      attr_127.create().set(m_attr_127);
      attr_128.create().set(m_attr_128);
      attr_129.create().set(m_attr_129);
      attr_130.create().set(m_attr_130);
      attr_131.create().set(m_attr_131);
      attr_132.create().set(m_attr_132);
      attr_133.create().set(m_attr_133); 
      attr_135.create().set(m_attr_135);    
      
      // POLARIZATION         

      Attribute<string> attr_137(polarization_grp, "GROUPE_TYPE");
      Attribute<string> attr_138 (polarization_grp, "COORDINATE_TYPE");
      Attribute< vector<string> > attr_139(polarization_grp, "STORAGE_TYPE");
      Attribute<int> attr_140(polarization_grp, "NOF_AXES");
      Attribute< vector<string> > attr_141(polarization_grp, "AXIS_NAMES");
      Attribute< vector<string> > attr_142(polarization_grp, "AXIS_UNIT");
      
      attr_137.value = m_attr_137;
      attr_138.value = m_attr_138;
      attr_139.create().set(m_attr_139);
      attr_140.value = m_attr_140;
      attr_141.create().set(m_attr_141);
      attr_142.create().set(m_attr_142);
	
  }	

	   
 
  
  
  
  